src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/TensorsInfo.cs

```diff
@@ -348,15 +464,18 @@ protected virtual void Dispose(bool disposing)
         
         private void CheckIndexBoundary(int idx)
         {
-            if (idx < 0 || idx >= _infoList.Count) {
-                throw new IndexOutOfRangeException("Invalid index [" + idx + "] of the tensors");
+            if (idx < 0 || idx >= _infoList.Count)
+            {
+                string msg = "Invalid index [" + idx + "] of the tensors";
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, msg);
             }
         }
 
         private class TensorInfo
```

Comment:
Why it private class?

Reply:
As I mentioned before, we supports multiple platform as almost same style APIs. In case of C language, supporting `TensorInfo` is hard to implement since `TensorsInfo` internally maintains them. Moreover, we do not want to deprecate previous APIs.

One of the possible solutions is that:
1) open TensorInfo as public
2) Overloading `AddTensorInfo(TensorInfo info)`

In my opinion, this way is not differ from previous one.
(i.e. `AddTensorInfo(string name, TensorType type, int[] dimension`)